pkg/openstack/instancesv2.go

```diff
@@ -79,6 +79,18 @@ func (os *OpenStack) instancesv2() (*InstancesV2, bool) {
 
 // InstanceExists indicates whether a given node exists according to the cloud provider
 func (i *InstancesV2) InstanceExists(ctx context.Context, node *v1.Node) (bool, error) {
+	if i.regionProviderID {
+		if node.Spec.ProviderID == "" {
+			klog.V(4).Infof("Instance %s should initialized first", node.Name)
+			return true, nil
```

Comment:
I haven't checked the call-sites, but my initial thought is that this should return an error rather than true. It feels like we are potentially asserting that the instance exists when in fact we don't know.

Reply:
cloud-provider does nothing if we return positive (it's alive and exist) response. Otherwise it delete/taint the node. The CCM in another region can make right decision. 

Than ProviderID is empty - yep, better return an error. 
If ProviderID is not openstack magic string - better do nothing. Errors will create noisy logs. 
AzureCCM does the same.

Reply:
This providerID topic was discussed recently, it seems there is the assumptions for cloud providers to set it, so new nodes will wait until the providerID is set unless the cloud provider indicates is not implemented ,https://github.com/kubernetes/kubernetes/pull/123713

Otherwise the node is not going to be initialized